### DISCUSSION TITLE
Add option to format upper/lower case only keywords, or only object names

### DISCUSSION THREAD

**Discussion by lukaseder** [dbeaver/dbeaver#565] [Ideas]
The current upper/lower case formatting option will just modify the entire selected string. Sometimes that's not what I want as a user. Usually, I'd like to just upper-case all keywords, as in:

``` sql
-- before
select 1 "a"

-- after
SELECT 1 "a"
```

But the identifiers / object names should stay the same. Specifically those that are put in double quotes (or brackets, backticks, depending on the database) should not be modified, because the result would be wrong.

So, there are two ways to tackle this:
1. Add additional formatting options to only upper/lower case all keywords and leave identifiers untouched.
2. Change the existing behaviour and modify only unquoted keywords / names, and leave quoted names untouched.

**Comment by serge-rider** [comment]
Currently DBeaver doesn't convert identifiers case at all. The only exception is identifiers auto-complete (database case is used for that).
So I don't see any problem. And I don't think that identifiers case should be modified ever.

**Comment by lukaseder** [comment]
Hmm, not sure what you mean. Consider this:

``` sql
select 1 "a"
```

Now select it, right click > format > to upper case and I'm getting this:

``` sql
SELECT 1 "A"
```

When I would have preferred to get this instead:

``` sql
SELECT 1 "a"
```

**Comment by serge-rider** [comment]
Case conversion is a very simple function which converts case of all letters. This function is a standard text editor action provided by Eclipse. It doesn't know anything about SQL.
To perform SQL formatting use "Format SQL" (ctrl+alt+f) action.

**Comment by lukaseder** [comment]
OKOK, let's _assume_ I find the standard Eclipse provided feature useless (because it's wrong to convert `"a"` to `"A"`). But I want to normalise all my keywords because that would be very nice. And I _don't_ want DBeaver to format my SQL statement, which I have formatted so nicely, manually with indentation and all.

Do you agree that it would be an interesting feature to just upper-case / lower-case all keywords (and perhaps, unquoted names)?

If it helps, that's how Oracle's SQL Developer works, for instance.

**Comment by serge-rider** [comment]
Agreed, that makes sense.
Although it may lead to UI over-complexity. Currently we have
- SQL formatter
- Case convertor (upper/lower)

This will be the third way to format SQL. Making two additional actions for case conversion with syntax parser looks too sophisticated. Maybe it'd be better to improve SQL formatter/add extra options.
WDYT?

**Comment by serge-rider** [comment]
After all I think that this feature can be covered by some external formatter (which now can be configured in preferences).
Making too many different formatters in DBeaver will just confuse users.

**Comment by Dian8** [comment]
Looks like already converts... Futhermore keywords are converted to lovercase. How to turn this feature off?

**Comment by serge-rider** [comment]
Preferences->General->Editors->SQL Editor->SQL Formatting->Auto format

**Comment by Dian8** [comment]
great! thanks!

**Comment by valentinmihaylov** [comment]
> The only exception is identifiers auto-complete (database case is used for that).

How can I change the "database case" to low-case?

**Comment by 18549** [comment]
Being able to auto format keywords to lowercase is the best thing ever.
